dev_guide/application_lifecycle/promoting_applications.adoc

```diff
@@ -0,0 +1,781 @@
+
+* If different image registries are used to assert isolation between a test
+environment and a production environment.
+* If different image repositories are used to separate test and production-ready
+images.
+
+If either of these are the case, the image stream must be modified when it is
+copied from the source environment to the target environment so that it resolves
+to the correct image. This is in addition to performing the steps described
+below to copy the image from one registry and repository to another.
+
+[[dev-guide-promoting-applications-methods-tools-summary]]
+=== Summary
+
+At this point, the following have been defined:
+
+* New application artifacts that make up a deployed application.
+* Correlation of application promotion activities to tools and concepts provided by {product-title}.
+* Integration between {product-title} and the CI/CD pipeline engine Jenkins.
+
+Putting together examples of application promotion flows within {product-title}
+is the final step for this topic.
+
+[[dev-guide-promoting-applications-scenarios]]
+== Scenarios and Examples
+
+Having defined the new application artifact components introduced by the Docker,
+Kubernetes, and {product-title} ecosystems, this section covers how to
+promote those components between environments using the mechanisms and
+tools provided by {product-title}.
+
+Of the components making up an application, the image is the primary artifact of
+note. Taking that premise and extending it to application promotion, the core,
+fundamental application promotion pattern is image promotion, where the unit of
+work is the image. The vast majority of application promotion scenarios entails
+management and propagation of the image through the promotion pipeline.
+
+Simpler scenarios solely deal with managing and propagating the image through
+the pipeline. As the promotion scenarios broaden in scope, the other application
+artifacts, most notably the API objects, are included in the inventory of items
+managed and propagated through the pipeline.
+
+This topic lays out some specific examples around promoting images as well as
+API objects, using both manual and automated approaches. But first, note the
+following on setting up the environment(s) for your application promotion
+pipeline.
+
+[[dev-guide-promoting-applications-setting-up]]
+=== Setting up for Promotion
+
+After you have completed development of the initial revision of your
+application, the next logical step is to package up the contents of the
+application so that you can transfer to the subsequent staging environments of
+your promotion pipeline.
+
+. First, group all the API objects you view as transferable and apply a common
+`label` to them:
++
+----
+labels:
+  promotion-group: <application_name>
+----
++
+As previously described, the `oc label` command facilitates the management of
+labels with your various API objects.
++
+[TIP]
+If you initially define your API objects in a {product-title} template, you can
```

Comment:
:confused: This is not in ==== format, but seems to be rendering fine in the preview build. Interesting!